### PR TITLE
Correctly configure CSC trigger primitive emulator for Run2 (81X)

### PR DESCRIPTION
**Pull request by dildick**
@ptcox @jrorie @tahuang @pakhotin

This pull request corrects known issues in the configuration of the CSC trigger primitive emulator for Run2. The era customizations should be cleaned up as well (it's a bit of a mess). Allow me to do this in another pull request. Fixing the settings is more urgent right now. 

I validated these changes by running on 2016B data using /store/user/abrinke1/EMTF/Run2016B/2016_04_29/RAW/ZeroBias1/272012/. 

I compared the BX of the LCT types:
- simCscTriggerPrimitiveDigis: emulated RAW data
- simCscTriggerPrimitiveDigis_MPCSORTED: emulated RAW data after sorting in the MPC
- csctfDigis: unpacked RAW data from CSC TrackFinder
- muonCSCDigis: : unpacked RAW data from Muon TrackFinder

I tuned the configuration so that simCscTriggerPrimitiveDigis_MPCSORTED would correspond better to csctfDigis. Results are shown below:

1) situation as it is in CMSSW 81X
comparison_lct_bx_2016b_original

2) after the changes from https://github.com/cms-sw/cmssw/pull/14391/commits/f1d5d4c4af8570a2bc36039171023db33c2de152
comparison_lct_bx_2016b_mpclct18_nosorting_nosmart

3) after shifting the BX of simCscTriggerPrimitiveDigis and simCscTriggerPrimitiveDigis_MPCSORTED by 2 to compare with csctfDigis
comparison_lct_bx_2016b_mpclct18_nosorting_nosmart_bxshift2

4) after the changes up to https://github.com/cms-sw/cmssw/pull/14391/commits/3e1dbd92393250918cc041d36c3495971d0e5528. Again, the BX of simCscTriggerPrimitiveDigis and simCscTriggerPrimitiveDigis_MPCSORTED is downshifted by 2 to compare with csctfDigis - the actual readout window is thus [5,11] for data LCTs. As can be seen the agreement between data and emulator is ~1.5%.
comparison_lct_bx_2016b_mpclct18_nosorting_nosmart_bxshift2_changereadout5to11_v2

### CODE REVIEW

File: L1Trigger/CSCTriggerPrimitives/python/cscTriggerPrimitiveDigisPostLS1_cfi.py [hunk 1]
```diff
@@ -1,5 +1,10 @@
 import FWCore.ParameterSet.Config as cms
 
+print " ######################################################################################################### "
+print " # WARNING: the module L1Trigger.CSCTriggerPrimitives.cscTriggerPrimitiveDigisPostLS1_cfi is deprecated. # "
```
Comment (slava77) [inline L1Trigger/CSCTriggerPrimitives/python/cscTriggerPrimitiveDigisPostLS1_cfi.py]: I think that all files including it in the release should be cleaned up, at least the standard configurations/sequences.
We have enough noise already during configuration building.

**Comment by slava77** [comment]
@cmsbuild please test

File: L1Trigger/CSCTriggerPrimitives/python/cscTriggerPrimitiveDigisPostLS1_cfi.py (hunk 1 above)
Comment (dildick) [inline L1Trigger/CSCTriggerPrimitives/python/cscTriggerPrimitiveDigisPostLS1_cfi.py]: @slava77 You're right. The configuration of L1Trigger/CSCTriggerPrimitives needs a serious clean-up. I will do that next week.

**Comment by dildick** [comment]
Run2 MPC does not drop invalid or low quality stubs.

**Comment by rekovic** [comment]
@dildick  Is there an analogous PR for for CMSSW_8_0_X as well?

**Comment by dildick** [comment]
@rekovic not yet

**Comment by fwyzard** [comment]
I've added a couple of comments in the 80x PR: #14419 - I can copy them here if you prefer.

**Comment by dildick** [comment]
It's fine. I'll update both PRs asap.

**Comment by rekovic** [comment]
+1

**Comment by dildick** [comment]
@rekovic Can you also sign #14419, please? Thank you.

**Comment by smuzaffar** [comment]
@dildick , one unit test is failing in IBs due to these changes. Can you please provide a fix for this

https://cmssdt.cern.ch/SDT/cgi-bin/buildlogs/slc6_amd64_gcc530/CMSSW_8_1_X_2016-05-16-1100/unitTestLogs/RecoMET/METProducers
```
----- Begin Fatal Exception 16-May-2016 21:17:23 CEST-----------------------
An exception of category 'Configuration' occurred while
   [0] Constructing the EventProcessor
   [1] Constructing module: class=CSCTriggerPrimitivesProducer label='simCscTriggerPrimitiveDigis'
Exception Message:
MissingParameter: ParameterSet 'mpcRun2' not found.
----- End Fatal Exception -------------------------------------------------
```

**Comment by dildick** [comment]
@smuzaffar Very strange. This should have shown up in all integration tests. The pset is clearly defined here https://github.com/dildick/cmssw/blob/53cf53ad6a34c9c4ebee00f5ef022ce29b366489/L1Trigger/CSCTriggerPrimitives/python/cscTriggerPrimitiveDigis_cfi.py#L574. How do I reproduce that particular unit test?

**Comment by smuzaffar** [comment]
@dildick , you can run the following commands to reproduce it
```
scram p CMSSW_8_1_X_2016-05-16-1100
cd CMSSW_8_1_X_2016-05-16-1100/
cmsenv
git cms-addpkg RecoMET/METProducers
scram b -j 8
scram b runtests
```

Looks like tests ran by https://github.com/cms-sw/cmssw/blob/CMSSW_8_1_X/RecoMET/METProducers/test/runtests.sh are not including the right python module? I see the warnings like
```
# WARNING: the module L1Trigger.CSCTriggerPrimitives.cscTriggerPrimitiveDigisPostLS1_cfi is deprecated. 
# Please import L1Trigger.CSCTriggerPrimitives.cscTriggerPrimitiveDigis_cfi
```

**Comment by dildick** [comment]
@smuzaffar That is correct. Those messages should be appearing. The file cscTriggerPrimitiveDigisPostLS1_cfi is not supposed to be used any longer and will be removed soon. The recommended one is cscTriggerPrimitiveDigis_cfi. It seems that the unit test is not configured properly for Run2 era customizations. I see messages as 

<PRE>
        #
        # -- Warning! You are using a deprecated customisation function. --
        #
        # It will probably run fine, but the customisations you are getting may be out of date.
        # You should update your configuration file by
        #   If using cmsDriver:
        #       1) remove the "--customise SLHCUpgradeSimulations/Configuration/postLS1Customs.customisePostLS1" option
        #       2) add the option "--era Run2_25ns" 
        #   If using a pre-made configuration file:
        #       1) remove or comment out the "process = customisePostLS1(process)" line.
        #       2) add "from Configuration.StandardSequences.Eras import eras" to the TOP of the config file (above
        #          the process declaration).
        #       3) add "eras.Run2_25ns" as a parameter to the process object, e.g. "process = cms.Process('HLT',eras.Run2_25ns)" 
        #
        # There is more information at https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideCmsDriverEras
        #
</PRE>

In this case the configuration does not run fine.

**Comment by dildick** [comment]
The cmsDriver command
<PRE>
cmsDriver.py TTbar_13TeV_TuneCUETP8M1_cfi  --conditions auto:run2_mc --fast  -n 1 --eventcontent FEVTDEBUGHLT -s GEN,SIM,RECOBEFMIX,DIGI:pdigi_valid,L1,L1Reco,RECO --beamspot NominalCollision2015 --customise SLHCUpgradeSimulations/Configuration/postLS1Customs.customisePostLS1 --fileout ttbarForMetTests.root
</PRE>
is out of date. Whoever is reponsible for that needs to update it.

**Comment by smuzaffar** [comment]
@lveldere and @TaiSakuma , can you please look in to this?

**Comment by smuzaffar** [comment]
@lveldere and @TaiSakuma  can you look in to this? 
@dildick , @davidlange6  , if you know how to fix it then let me know (I can update it)

**Comment by slava77** [comment]
remove ```--customise .... ```
add ``` --era Run2_2016```

**Pull request merged**